ray-operator/apis/ray/v1alpha1/rayservice_types.go

```diff
@@ -23,22 +23,25 @@ const (
 // RayServiceSpec defines the desired state of RayService
 type RayServiceSpec struct {
 	// Important: Run "make" to regenerate code after modifying this file
+	ServeDeploymentGraphSpec ServeDeploymentGraphSpec `json:"serveDeploymentGraphConfig,omitempty"`
+	RayClusterSpec           RayClusterSpec           `json:"rayClusterConfig,omitempty"`
+}
+
+type ServeDeploymentGraphSpec struct {
+	ImportPath       string            `json:"importPath"`
+	RuntimeEnv       string            `json:"runtimeEnv,omitempty"`
 	ServeConfigSpecs []ServeConfigSpec `json:"serveConfigs,omitempty"`
-	RayClusterSpec   RayClusterSpec    `json:"rayClusterConfig,omitempty"`
 }
 
 // ServeConfigSpec defines the desired state of RayService
 // Reference to https://docs.ray.io/en/latest/ray-core/package-ref.html#ray-remote.
 type ServeConfigSpec struct {
 	Name                      string             `json:"name"`
-	ImportPath                string             `json:"importPath"`
-	InitArgs                  []string           `json:"initArgs,omitempty"`
-	InitKwargs                map[string]string  `json:"initKwargs,omitempty"`
 	NumReplicas               *int32             `json:"numReplicas,omitempty"`
 	RoutePrefix               string             `json:"routePrefix,omitempty"`
 	MaxConcurrentQueries      *int32             `json:"maxConcurrentQueries,omitempty"`
-	UserConfig                map[string]string  `json:"userConfig,omitempty"`
-	AutoscalingConfig         map[string]string  `json:"autoscalingConfig,omitempty"`
+	UserConfig                string             `json:"userConfig,omitempty"`
```

Comment:
Are the dictionary types string-ified for the operator?

Reply:
Here we let users to input string and we will convert string to dict in ray-operator/controllers/ray/utils/serve_httpclient.go:159

Reply:
How does the user get the string? Do they have to manually construct a dictionary in Python and then do `json.dumps()` to get it? Or could they do something like:

```yaml
UserConfig: {
    key_1: val_1,
    key_2: val_2,
}
```
directly in the K8s config?

Reply:
In the config it will be 
```
| UserConfig: {
    key_1: val_1,
    key_2: val_2,
}
```

In python, I think it is json.dumps() which convert dict to json string

Reply:
Got it, that sounds good.